src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -209,12 +245,6 @@ int main(int argc, char *argv[])
     Log::Info << "Loaded kNN model from '" << inputModelFile << "' (trained on "
         << knn.Dataset().n_rows << "x" << knn.Dataset().n_cols << " dataset)."
         << endl;
-
-    // Adjust singleMode and naive if necessary.
-    knn.SingleMode() = CLI::HasParam("single_mode");
-    knn.Naive() = CLI::HasParam("naive");
-    knn.LeafSize() = size_t(lsInt);
-    knn.Epsilon() = epsilon;
```

Comment:
@rcurtin
Ok, I understand.
I updated the code to consider `--single_mode`, `--naive`, `--leaf_size`, and  `--epsilon`, and update that values in the input model (https://github.com/mlpack/mlpack/pull/747/commits/ecaf1bf80976b83044cf9d90b65aa9a02e4efba1).
The leaf size specified will be only considered for que query tree. I added a new comment about this (because it doesn't make sense to rebuild the input model file). Would you agree?
If you agree, I will add the `--mode` option once the PR of Greedy Search is merged.
Thanks!